Feature/93 redo in rust

cainsusk opened the pull request:
fully implements the user (backend and frontend) as well as the api's for transaction and listing (backend).

Thus, the backend is fully implemented. There may still be some bugs, however.

closes #86

cainsusk committed:
327-93 Fully implement db <-> backend <-> frontend

cainsusk committed:
327-93 Implement listing api and almost finish booking

Adds the api for creating and updating a listing as well as
brings the booking module close to completion.

cainsusk committed:
327-93 Implement transaction api

completes the implementation of the transaction api, making sure no
booking can be made within the range of another booking on the same
listing.

cainsusk committed:
327-93 Implement listing api

Fully implements listing api, with descriptions and time of last edit
included.

cainsusk committed:
327-93 Implement listing and transaction front end

implements the pages for viewing, updating, and booking listings.

cainsusk committed:
327-93 Implement home page and account balance

adds user bookings to the home page as well as an account balance
system.

datmaxo reviewed (dismissed):
looks awesome, thank you so much for going through the effort of rebuilding the systems!

cainsusk committed:
327-93 Restructure Project

Moves the rust code out of the alt directory and moves the java code to
the archive directory.
